Tests/AppTests/Controllers/User/UserPhotoTest.swift

```diff
@@ -0,0 +1,206 @@
+import XCTest
+import Testing
+import HTTP
+import Core
+import Foundation
+import Multipart
+import FormData
+import Crypto
+
+@testable import Vapor
+@testable import App
+
+// swiftlint:disable superfluous_disable_command
+// swiftlint:disable force_try
+class UserPhotoTest: TestCase {
+
+  let photoPath = "user_photos/1/"
+  let testResouces = "Tests/Resources/"
+  var storedDir = ""
+  let fileManager = Foundation.FileManager()
+  var clientToken = ""
+  var headers: [HeaderKey: String] = [:]
+  var domain = ""
+
+  override func setUp() {
+    super.setUp()
+    do {
+      try drop.truncateTables()
+      storedDir = drop.config.publicDir + photoPath
+      clientToken = drop.config["app", "client-token"]?.string ?? ""
+      domain = drop.config["app", "domain"]!.string!
+      headers = [
+        "client-token": clientToken,
+        HeaderKey.contentType: TestConstants.Header.Value.multipartFormData
+      ]
+    } catch {
+      XCTFail("Droplet set raise exception: \(error.localizedDescription)")
+      return
+    }
+  }
+
+  func testThatUserPhotoIsUpdatedByFormDataSend() throws {
+
+    let user = User()
+    user.photo = nil
+    try! user.save()
+
+    let fileName = "robot-1469114466kSY.jpg"
+    let filePath = drop.config.workDir + testResouces + fileName
+    let storedFilePath = storedDir + fileName
+
+    let expectedPhoto = "\(domain)/\(photoPath + fileName)"
+
+    try! fileManager.removeAllFiles(atPath: storedDir)
+
+    let filePhoto = try! Vapor.FileManager.readBytesFromFile(filePath)
+    let partPhoto = Part(headers: [:], body: filePhoto)
+    let photo = Field(name: "photo", filename: fileName, part: partPhoto)
+
+    headers[HeaderKey.authorization] = "Bearer \(try user.token())"
+
+    let request = Request(method: .patch, uri: "http://0.0.0.0:8080/user/1", headers: headers)
+    request.formData = ["photo": photo]
+    
+    let response = try drop.respond(to: request)
+    response.assertStatus(is: .ok)
+
+    guard
+      let updatedUser = response.json,
+      let newPhoto = updatedUser["photo"]?.string
+    else {
+      XCTFail("Can't get updatedUser")
+      return
+    }
+
+    XCTAssertEqual(expectedPhoto, newPhoto, "got: '\(newPhoto)', expected: '\(expectedPhoto)'")
+    XCTAssertTrue( try! CryptoHasher.compareFiles(filePath1: filePath, filePath2: storedFilePath))
+  }
+
+  func testThatUserPhotoIsUpdatedByPhotoSendAsBase64EncodedString() throws {
```

Comment:
may be `...UsingPhotoWithBase64EncodedStringContent`

Reply:
ok, fixed